jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2046,9 +2046,7 @@ public String getName() {
 		return name;
 	}
 	//must have been null to set it
-	name = VM.getClassNameImpl(this).intern();
-	classNameString = name;
-	return name;
+	return VM.getClassNameImpl(this);
```

Comment:
Have you checked the other callers of this in Java?  In both cases, I see comment indicating the class names shouldn't be interned:
```
/*[PR 126182] Do not intern bootstrap class names when loading */
```
(though this may have already been violated by the stackwalkers storing interned names on the class)

Reply:
We'll have to see if we can find the PR - all testing passes with this change, so I don't believe there's a functional issue.

Reply:
We'll need to check the footprint as interning all bootstrap class names makes them immortal.  Lot's of classes that probably never have their names checked apart from say, the classloading code

Reply:
@pshipton Are you able to find the referenced PR or recall why we would have a difference in behaviour here? Perhaps it was a performance issue rather than correctness.

Reply:
The code in question is:
https://github.com/eclipse-openj9/openj9/blob/df51c203a8b47e8f81df4d0f29c5e1e272771f43/jcl/src/java.base/share/classes/com/ibm/oti/vm/AbstractClassLoader.java#L172-L181

Reply:
> /*[PR 126182] Do not intern bootstrap class names when loading */ 

This appears a bugzilla PR, the server is long gone I believe.
Note: `com.ibm.oti.vm.AbstractClassLoader` is only used by Java 8.

Reply:
I've restored the original behaviour for java 8. Since all of the callers of the new code are already GC-aware, I'm going to leave the VMHelper the way it is.